src/game/MotionGenerators/MovementHandler.cpp

```diff
@@ -349,6 +349,14 @@ void WorldSession::HandleMovementOpcodes(WorldPacket& recv_data)
             mover = _player;
     }
 
+    if (opcode == CMSG_MOVE_SET_FLY)
+    {
+        if (plMover)
+            if (Pet* pet = plMover->GetPet())
+                if (pet->isControlled() && plMover->HasAuraType(SPELL_AURA_MOD_FLIGHT_SPEED_MOUNTED))
```

Comment:
Cheaper to check ismounted and MOVEFLAG_CAN_FLY

Reply:
I think the moveflag_can_fly is redundant in this case, because `CMSG_MOVE_SET_FLY` already implies that a takeoff is happening, so it can't happen without that moveflag anyway. I'll just check for IsMounted